boxsdk/auth/oauth2.py

```diff
@@ -240,17 +242,40 @@ def send_token_request(self, data, access_token, expect_refresh_token=True):
             url,
             data=data,
             headers=headers,
-            access_token=access_token
+            access_token=access_token,
         )
         if not network_response.ok:
             raise BoxOAuthException(network_response.status_code, network_response.content, url, 'POST')
         try:
             response = network_response.json()
-            self._access_token = response['access_token']
-            self._refresh_token = response.get('refresh_token', None)
-            if self._refresh_token is None and expect_refresh_token:
+            access_token = response['access_token']
+            refresh_token = response.get('refresh_token', None)
+            if refresh_token is None and expect_refresh_token:
                 raise BoxOAuthException(network_response.status_code, network_response.content, url, 'POST')
         except (ValueError, KeyError):
             raise BoxOAuthException(network_response.status_code, network_response.content, url, 'POST')
-        self._store_tokens(self._access_token, self._refresh_token)
+        self._store_tokens(access_token, refresh_token)
         return self._access_token, self._refresh_token
+
+    def revoke(self):
+        """
+        Revoke the authorization for the current access/refresh token pair.
+        """
+        with self._refresh_lock:
+            token_to_revoke, _ = self._get_tokens()
+            if token_to_revoke is None:
+                return
+            url = '{base_auth_url}/revoke'.format(base_auth_url=API.OAUTH2_API_URL)
+            network_response = self._network_layer.request(
+                'POST',
+                url,
+                data={
+                    'client_id': self._client_id,
+                    'client_secret': self._client_secret,
+                    'token': token_to_revoke,
+                },
+                access_token=token_to_revoke,
+            )
+            if not network_response.ok:
+                raise BoxOAuthException(network_response.status_code, network_response.content, url, 'POST')
+            self._store_tokens(None, None)
```

Comment:
Idea: add auto-revoking features:

```python
def close(self):
    return self.revoke()

def __enter__(self):
    return self

def __exit__(self, *exc_info):
    self.close()

def __del__(self):
    if self._revoke_tokens_on_del:
        self.close()
```

where `_revoke_tokens_on_del` is set from an ``__init__`` param `revoke_tokens_on_del` (defaulting to `False`).

Then, for applications that never want to leave active tokens around, they can be run like

```python
with OAuth2(*args, **kwargs) as oauth2:
    ...
```

or

```python
oauth2 = OAuth2(*args, revoke_tokens_on_del=True, **kwargs)
```